airbyte/caches/base.py

```diff
@@ -957,6 +958,11 @@ def register_source(
         This method is called by the source when it is initialized.
         """
         self._source_name = source_name
+        self.file_writer.register_source(
+            source_name,
+            incoming_source_catalog,
+            stream_names=stream_names,
+        )
```

Comment:
I don't know if file writers were getting the source registered and expected stream names initialized. This change makes sure when the cache gets its source registered, the file writer does also.